demoapp/demo_app.py

```diff
@@ -58,224 +58,430 @@ def get_mgl_sections_file() -> pd.DataFrame:
+            Response Format:\nSummary:\nSUMMARY\nCategory:\nCATEGORY\nTags:\nTAGS")
+            """
     with get_openai_callback() as cb:
         response = rag_chain.invoke(query)
         st.write(cb.total_tokens, cb.prompt_tokens, cb.completion_tokens, cb.total_cost)
     return response
+       
+def generate_response_small_documents(bill_title: str, bill_text: str, mgl_ref: str, mgl_names: List[Dict[str, str]], llm: object) -> str:
+    """
+    Generates a response for small documents by summarizing a bill and its relevance to Mass General Laws.
+    
+    Args:
+        bill_title (str): Title of the bill.
+        bill_text (List[str]): List of text segments from the bill.
+        mgl_ref (List[str]): List of Mass General Law references.
+        mgl_names (List[Dict[str, str]]): List of dictionaries containing names of the chapters and sections of Mass General Laws.
+        llm (Any): Language model used for generating responses.
 
+    Returns:
+        Any: The response from the language model based on the generated prompt.
+    """
 
-#Function to update or append to CSV
-# def update_csv(bill_num, title, summarized_bill, category, tag, csv_file_path, rouge_scorer, cosine_sim_score, fact_const_score ):
-#     try:
-#         df = pd.read_csv(csv_file_path)
-#     except FileNotFoundError:
-#         # If the file does not exist, create a new DataFrame
-#         df = pd.DataFrame(columns=["Bill Number", "Bill Title", "Summarized Bill", "Category", "Rouge-1", "Rouge-2", "Rouge-L, Cosine Similarity, Fact Consistency"])
-#     mask = df["Bill Number"] == bill_num
-#     if mask.any():
-#         df.loc[mask, "Bill Title"] = title
-#         df.loc[mask, "Summarized Bill"] = summarized_bill
-#         df.loc[mask, "Category"] = category
-#         # df.loc[mask, "Tags"] = tag
-#         df.loc[mask, "Rouge-1"] = f"{rouge_scorer['rouge1'].fmeasure:.2f}"
-#         df.loc[mask, "Rouge-2"] = f"{rouge_scorer['rouge2'].fmeasure:.2f}"
-#         df.loc[mask, "Rouge-L"] = f"{rouge_scorer['rougeL'].fmeasure:.2f}"
-#         df.loc[mask, "Cosine Similarity"] = cosine_sim_score
-#         df.loc[mask, "Fact Consistency"] = fact_const_score
+    mgl_documents = [Document(page_content=x) for x in mgl_ref]
+  
+    try:
+        bill_text = [Document(page_content=x) for x in bill_text]
+    except Exception as e:
+        print("No Bill Text")
         
-#     else:
-#         new_bill = pd.DataFrame(columns=["Bill Number", "Bill Title", "Summarized Bill", "Category", "Rouge-1", "Rouge-2", "Rouge-L, Cosine Similarity, Fact Consistency"])
-#         df = pd.concat([df, new_bill], ignore_index=True)
     
-#     df.to_csv(csv_file_path, index=False)
-#     return df
+    if mgl_documents is None or str(mgl_documents).strip().lower()  == "nan" or mgl_documents == "":
```

Comment:
Is the bill text being inserted as '{context}'? If so, I have a concern that it will not be separated distinctly enough for the LLM to distinguish clearly from the prompt. Could a format like this work better?

```
Can you please explain what the following MA bill means to a regular resident without specialized knowledge? Please provide a one paragraph summary in 4 sentences. 
Please be simple, direct and concise for the busy reader. Make bullet points if possible.
After generating summary, classify the bill according to the list of categories below to the closest relevant category. Do not output categories outside of the list. 
Then, output the top 3 tags in this specific category from the list of tags that are relevant to this bill. Do not output the tags outside from the list. 
Note that the bill refers to specific existing chapters and sections of the Mass General Laws. Use the corresponding names of Mass General Law chapter and sections for constructing your summary, generating categories and tags.

Response Format:
Summary: [SUMMARY]
Category:[CATEGORY]
Tags: [TAGS]

The bill title is: """{title}"""

The bill text is: """{bill_text}"""

The relevant section names are: """{names}"""

The relevant section text is: """{mgl_sections}"""

The set of categories is: """{categories}"""

The set of tags is: """{tags}"""
```

Note I recommend using 'bill_text' instead of 'context' for clarity.

Reply:
I bumped this concern in slack

Reply:
Unfortunately, it seems like you have to "context". I tried using something else before, but it wasn't working. I haven't figured out a way around that. Here is the issue: https://github.com/langchain-ai/langchain/discussions/16001